components/TextFields/tests/snapshot/MDCBaseTextFieldSnapshotTests.m

```diff
@@ -18,21 +18,50 @@
 
 #import "MaterialTextFields+ContainedInputView.h"
 
+static const NSTimeInterval kTextFieldValidationAnimationTimeout = 1.0;
+
 @interface MDCBaseTextFieldTestsSnapshotTests : MDCSnapshotTestCase
+@property(strong, nonatomic) MDCBaseTextField *textField;
 @end
 
 @implementation MDCBaseTextFieldTestsSnapshotTests
 
 - (void)setUp {
   [super setUp];
 
+  self.textField = [self createBaseTextFieldInKeyWindow];
   // Uncomment below to recreate all the goldens (or add the following line to the specific
   // test you wish to recreate the golden for).
   //    self.recordMode = YES;
 }
 
 - (void)tearDown {
   [super tearDown];
+  [self.textField removeFromSuperview];
+  self.textField = nil;
+}
+
+- (MDCBaseTextField *)createBaseTextFieldInKeyWindow {
+  MDCBaseTextField *textField = [[MDCBaseTextField alloc] initWithFrame:CGRectMake(0, 0, 200, 60)];
+  textField.borderStyle = UITextBorderStyleRoundedRect;
+  UIWindow *keyWindow = [[UIApplication sharedApplication] keyWindow];
+  [keyWindow addSubview:textField];
+  return textField;
+}
+
+- (void)validateTextField:(MDCBaseTextField *)textField {
```

Comment:
FYI: It seems to be very similar to what you have going on here. 

https://github.com/material-components/material-components-ios/blob/5bee8bc3b6be6745b777fbc091131489da4fb4d0/components/TextFields/tests/snapshot/supplemental/MDCTextFieldSnapshotTestCase.m

We may want to copy some of the explanation for why this is needed.

Reply:
Which part do you want copied? I do have some explanation in there, I can add more in a follow up 👍